### PR TITLE
Make drizzle_binlog_get_filename return end position

### PR DESCRIPTION
**Pull request by andreas-bok-sociomantic**
Modify the function so the end position of the file is returned

### CODE REVIEW

**Comment by mathias-lang-sociomantic** [comment]
That's definitely not a point release. Actually it should even be a [major](https://community.kde.org/Policies/Binary_Compatibility_Examples#Change_the_return_type).

**Comment by andreas-bok-sociomantic** [comment]
> That's definitely not a point release. Actually it should even be a major.

Add to the API yes, but doesn't break anything.

**Comment by andreas-bok-sociomantic** [comment]
Would concede that it might should go into to a minor release.

**Comment by andreas-bok-sociomantic** [comment]
- Changed the function signature so it returns a drizzle_return_t` value
- Added unittest of the changed function signature 
- Updated RST documentation 
- Rebased on latest v6.x.x

**Comment by andreas-bok-sociomantic** [comment]
#103 should be merged first into v6.x.x

**Comment by mathias-lang-sociomantic** [comment]
> #103 should be merged first into v6.x.x

Still targeting v5.x.x though ?...

**Comment by andreas-bok-sociomantic** [comment]
> Still targeting v5.x.x though ?...

Fixed

**Comment by andreas-bok-sociomantic** [comment]
More comments?

**Comment by ben-palmer-sociomantic** [comment]
> More comments?

It is still blocked at the moment, so is it ready for review?

**Comment by andreas-bok-sociomantic** [comment]
> It is still blocked at the moment, so is it ready for review?

Yes

File: libdrizzle-5.1/binlog.h [hunk 1]
```diff
@@ -203,23 +203,32 @@ DRIZZLE_API
...
  * The filename parameter is allocated by the function and needs to be
- * freed by the application when finished with.
+ * freed by the application when finished with. This is the case, regardless of
+ * the return status of the function.
  *
- * @param[in] con Drizzle structure previously initialized with drizzle_create().
- * @param[in,out] filename buffer to copy filename to
- * @param[in] file_index index of the binlog to retrieve.
+ * @param[in] con Drizzle structure previously initialized with
+ * drizzle_create() or drizzle_clone().
```
Comment (ben-palmer-sociomantic) [inline libdrizzle-5.1/binlog.h]: No more `drizzle_clone` right?

File: libdrizzle-5.1/binlog.h [hunk 2]
```diff
@@ -203,23 +203,32 @@ DRIZZLE_API
...
+ * @param[in,out] filename Buffer to copy filename to
+ * @param[in,out] end_position Variable to save the size of the binlog file into
+ * @param[in] file_index: Index of the binlog to retrieve.
```
Comment (ben-palmer-sociomantic) [inline libdrizzle-5.1/binlog.h]: Most of the parameters don't have `:` in them.

**Review by ben-palmer-sociomantic — COMMENTED** [review]
I had a couple of comments on the documentation otherwise LGTM.

**Comment by andreas-bok-sociomantic** [comment]
Addressed comments by @ben-palmer-sociomantic 

Before merging: 
We might consider another change so the client is required to provide a preallocated buffer to copy   the filename into. Would make mem leaks less likely.
@mathias-lang-sociomantic

**Comment by ben-palmer-sociomantic** [comment]
> We might consider another change so the client is required to provide a preallocated buffer to copy the filename into. Would make mem leaks less likely.

Whats the way other methods in the library do it?

**Comment by andreas-bok-sociomantic** [comment]
> Whats the way other methods in the library do it?

[`drizzle_escape_str`](https://github.com/sociomantic-tsunami/libdrizzle-redux/blob/v5.6.x/libdrizzle/query.cc#L60) doesn't take a preallocated buffer. 
That's the only other API method with similar behavior.

**Comment by ben-palmer-sociomantic** [comment]
> drizzle_escape_str doesn't take a preallocated buffer. That's the only other API method with similar behavior.

In that case I am fine with keeping it as it is.

File: libdrizzle/binlog.cc [hunk 3]
```diff
@@ -449,44 +449,52 @@ const char *drizzle_binlog_event_type_str(drizzle_binlog_event_types_t event_typ
...
+    drizzle_log_info(con, __FILE_LINE_FUNC__, "Found binary log '%s' with filesize %d bytes",
+                     *filename, *end_position);
+  }
+  else
+  {
+    drizzle_log_info(con, __FILE_LINE_FUNC__, "No binary log files found");
   }
 
   drizzle_result_free(result);
+  return DRIZZLE_RETURN_OK;
```
Comment (ben-palmer-sociomantic) [inline libdrizzle/binlog.cc:501]: Is it possible that with `drizzle_result_free(result);` only being called here we could end up returning an error code without freeing the result?

**Comment by ben-palmer-sociomantic** [comment]
Just one more comment then LGTM.

File: libdrizzle/binlog.cc (hunk 3 above)
Comment (andreas-bok-sociomantic) [reply libdrizzle/binlog.cc:501]: Indeed, that would be the case. Will fix

**Comment by andreas-bok-sociomantic** [comment]
Addressed comment from @ben-palmer-sociomantic about possible memory leak. 

```diff
diff --git a/libdrizzle/binlog.cc b/libdrizzle/binlog.cc
index 9d30a5a..8a27b92 100644
--- a/libdrizzle/binlog.cc
+++ b/libdrizzle/binlog.cc
@@ -464,6 +464,7 @@ drizzle_return_t drizzle_binlog_get_filename(drizzle_st *con, char **filename,
   if (driz_ret != DRIZZLE_RETURN_OK)
   {
     drizzle_set_error(con, __FILE_LINE_FUNC__, "Query to retrieve binary logs failed");
+    drizzle_result_free(result);
     return driz_ret;
   }
 
@@ -473,6 +474,7 @@ drizzle_return_t drizzle_binlog_get_filename(drizzle_st *con, char **filename,
   if (file_index < -1 || file_index >= row_count)
   {
     drizzle_set_error(con, __FILE_LINE_FUNC__, "Invalid binlog file index: %d", file_index);
+    drizzle_result_free(result);
     return DRIZZLE_RETURN_INVALID_ARGUMENT;
   }
```

**Pull request merged**